### PR TITLE
Viewframe update

### PR DESCRIPTION
**Pull request by agalitsyna**
Cooltools transfer to bioframe.viewframe
Key tasks/issues resolved: 
- 1-5 from https://github.com/open2c/cooltools/issues/262
- https://github.com/open2c/cooltools/issues/255
- https://github.com/open2c/cooltools/issues/254
- https://github.com/open2c/cooltools/issues/257
- https://github.com/open2c/cooltools/pull/260

### CODE REVIEW

File: cooltools/dotfinder.py [hunk 1]
```diff
@@ -1341,7 +1342,9 @@ def scoring_step(
             # "create_cooler" API needs more consistency:
             # https://github.com/mirnylab/cooler/issues/149
             #  manually describing "data"-columns (except for bin_ids).
-            columns = ["count",] + ["la_exp." + k + ".value" for k in kernels]
+            columns = [
+                "count",
```
Comment (sergpolly) [inline cooltools/dotfinder.py:1346]: sometime `black` does strange things - like this one ...
I hope, at least , that it stays consistent from run to run, from version to version

File: tests/test_compartments_saddle.py [hunk 2]
```diff
@@ -120,12 +117,8 @@ def test_trans_saddle_cli(request, tmpdir):
...
     runner = CliRunner()
     result = runner.invoke(
-        cli, [
-            'call-compartments',
-            '--contact-type', "trans",
-            '-o', out_eig_prefix,
-            in_cool
-        ]
+        cli,
+        ["call-compartments", "--contact-type", "trans", "-o", out_eig_prefix, in_cool],
```
Comment (sergpolly) [inline tests/test_compartments_saddle.py:121]: `black` ...

File: tests/test_coverage.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import numpy as np
+
+
+def test_coverage_symmetric_upper(request):
+    # perform test:
+    clr = cooler.Cooler(op.join(request.fspath.dirname, "data/CN.mm9.1000kb.cool"))
+    cov = cooltools.coverage.get_coverage(clr, ignore_diags=2, chunksize=int(1e7))
+
+    # Test that minimal coverage is larger than 0.5
+    assert cov[cov > 0].min() >= 1
```
Comment (sergpolly) [inline tests/test_coverage.py:14]: the debate is still pending for this

**Review by sergpolly — CHANGES_REQUESTED** [review]
This looks great ! Should be ready to merge after we fix maybe a couple of tiny items:
1. `RuntimeError`  -> `ValueError` in a couple of places
2. remove some comments related to the issues with `regions, regions2` not being unique for trans and some asymmetric cases
3. resolve the debate regarding coverage - i.e. do we do `cov` or `cov/2` ?

We cannot do much with `black`, can we ? I feel like we ran `black` on the entire codebase before, but some weird formatting changes are still creeping in. I don't understand it fully. Maybe we could agree on doing `black` as separate PRs every now and then, to make the changes looks smaller ? Also , any chance we could have slightly different `black` parameters - so that we keep re-`blacking` over and over ?

File: cooltools/coverage.py [hunk 4]
```diff
@@ -44,8 +44,6 @@ def _get_chunk_coverage(chunk, pixel_weight_key="count"):
     covs[1] += np.bincount(pixels["bin1_id"], weights=pixel_weights, minlength=n_bins)
     covs[1] += np.bincount(pixels["bin2_id"], weights=pixel_weights, minlength=n_bins)
 
-    covs = covs / 2
```
[the remark this answers is not in the document]
Comment (gfudenberg) [reply cooltools/coverage.py:47]: discussion 9/7/21:
-- lets go with removing /2 everywhere, including the [special case]( https://github.com/open2c/cooltools/blob/ff066aca1cd1569c119cfe74efdb54beee580e20/cooltools/coverage.py#L111), and update the docstring to clarify that the sum( total_coverage) from this function is 2x the number of reads contributing to the cooler. The storage mode is apparently for a different type of coolers (e.g. RNA vs DNA) that are not supported currently by cooltools.
    
Calculate the sums of cis and genome-wide contacts (aka coverage aka marginals) for  a sparse Hi-C contact map in Cooler HDF5 format. Note that the sum( `tot_raw_cov`) from this function is two times the number of reads contributing to the cooler as each side contributes to the coverage.

File: tests/test_coverage.py (hunk 3 above)
Comment (sergpolly) [reply tests/test_coverage.py:14]: Based on the discussion 9/7/2021 -> this is correct

**Review by sergpolly — APPROVED** [review]

**Pull request merged**